src/libraries/System.Formats.Cbor/src/System/Formats/Cbor/Writer/CborWriter.cs

```diff
@@ -60,14 +62,42 @@ public partial class CborWriter
         /// <exception cref="ArgumentOutOfRangeException"><paramref name="conformanceMode" /> is not a defined <see cref="CborConformanceMode" />.</exception>
-        public CborWriter(CborConformanceMode conformanceMode = CborConformanceMode.Strict, bool convertIndefiniteLengthEncodings = false, bool allowMultipleRootLevelValues = false)
+        [EditorBrowsable(EditorBrowsableState.Never)]
+        public CborWriter(CborConformanceMode conformanceMode, bool convertIndefiniteLengthEncodings, bool allowMultipleRootLevelValues)
+            : this(conformanceMode, convertIndefiniteLengthEncodings, allowMultipleRootLevelValues, DefaultCapacitySentinel)
+        {
+        }
+
+        /// <summary>Initializes a new instance of <see cref="CborWriter" /> class using the specified configuration.</summary>
+        /// <param name="conformanceMode">One of the enumeration values that specifies the guidance on the conformance checks performed on the encoded data.
+        /// Defaults to <see cref="CborConformanceMode.Strict" /> conformance mode.</param>
+        /// <param name="convertIndefiniteLengthEncodings"><see langword="true" /> to enable automatically converting indefinite-length encodings into definite-length equivalents and allow use of indefinite-length write APIs in conformance modes that otherwise do not permit it; otherwise, <see langword="false" />.</param>
+        /// <param name="allowMultipleRootLevelValues"><see langword="true" /> to allow multiple root-level values to be written by the writer; otherwise, <see langword="false" />.</param>
+        /// <param name="initialCapacity">The initial capacity of the underlying buffer. The value -1 can be used to use the default capacity.</param>
+        /// <exception cref="ArgumentOutOfRangeException">
+        /// <para><paramref name="conformanceMode" /> is not a defined <see cref="CborConformanceMode" />.</para>
+        /// <para>-or-</para>
+        /// <para><paramref name="initialCapacity"/> is not zero, positive, or the default value indicator -1.</para>
+        /// </exception>
+        public CborWriter(
+            CborConformanceMode conformanceMode = CborConformanceMode.Strict,
+            bool convertIndefiniteLengthEncodings = false,
+            bool allowMultipleRootLevelValues = false,
+            int initialCapacity = DefaultCapacitySentinel)
         {
             CborConformanceModeHelpers.Validate(conformanceMode);
 
             ConformanceMode = conformanceMode;
             ConvertIndefiniteLengthEncodings = convertIndefiniteLengthEncodings;
             AllowMultipleRootLevelValues = allowMultipleRootLevelValues;
             _definiteLength = allowMultipleRootLevelValues ? null : (int?)1;
+
+            _buffer = initialCapacity switch
+            {
+                DefaultCapacitySentinel or 0 => null!,
```

Comment:
This align the field with its nullability annotation and remove the need for null checks in the resize method:
```suggestion
                DefaultCapacitySentinel or 0 => Array.Empty<byte>(),
```